### PR TITLE
Improve firefox support

### PR DESCRIPTION
**Pull request by mrmr1993**
This merges all of the code mentioned in [my comment](https://github.com/philc/vimium/issues/2425#issuecomment-294355905) on #2425.

As listed there:
* link hints for opening links anywhere *except* the current tab do not work
* clipboard operations do not work.

Also, there are still issues with focus in the find mode HUD.
(Branch [hud-focus](https://github.com/mrmr1993/vimium/tree/hud-focus) seems to improve the situation, by using separate persistent elements for normal HUD and the HUD find input. Not sure if this is a direction we want to take.)

### CODE REVIEW

File: content_scripts/mode_find.coffee [hunk 1]
```diff
@@ -178,7 +178,10 @@ class FindMode extends Mode
       # ignore the selectionchange event generated by find()
       document.removeEventListener("selectionchange", @restoreDefaultSelectionHighlight, true)
 
-    result = window.find(query, options.caseSensitive, options.backwards, true, false, true, false)
+    try
+      result = window.find(query, options.caseSensitive, options.backwards, true, false, true, false)
+    catch # Failed searches throw on Firefox.
+    options.postFindFocus?.focus()
```
Comment (smblott-github) [inline content_scripts/mode_find.coffee]: Is the indentation correct here?  Looks like `XXX.focus()` should happen only if the exception is raised.

File: pages/content_script_loader.coffee [hunk 2]
```diff
@@ -0,0 +1,28 @@
+injectContentScripts = ->
```
Comment (smblott-github) [inline pages/content_script_loader.coffee]: Is this file actually being used?

File: content_scripts/mode_find.coffee (hunk 1 above)
Comment (mrmr1993) [inline content_scripts/mode_find.coffee]: Yes. `window.find` in Firefox focuses the window that we call it on, so this gives us an opportunity to refocus the HUD window.

Should I add a newline between the `try ... catch` and this line, to clearly separate them?

File: pages/content_script_loader.coffee (hunk 2 above)
Comment (mrmr1993) [inline pages/content_script_loader.coffee]: No, sorry. The revert commit reintroduced it and I didn't notice. I'll remove it

**Comment by mrmr1993** [comment]
@smblott-github I've pushed some commits to address your comments.

**Comment by smblott-github** [comment]
@mrmr1993 OK.  Merged this.  I'm about to merge #2470 too (move to `keydown`, `event.key`).  I don't use Firefox, so the ball is largely in your court in terms of figuring out what works and what doesn't.

**Pull request merged**